### PR TITLE
Fix throwing exception in `as` outside of Task

### PR DESCRIPTION
**Pull request by wedens**

### CODE REVIEW

**Comment by wedens** [comment]
For some reason CI fails on 2.11.8. Can someone help me to understand what's wrong?

**Comment by bfritz** [comment]
@wedens Looks like the [test failure](https://travis-ci.org/http4s/http4s/jobs/223896387#L1673) is unrelated to your changes.  We've seen a few of those lately, and we're hoping #1123 squashes them.

**Comment by wedens** [comment]
It's not a semantic change. `Task.map` (used inside `EitherT[Task, ?, ?].fold`) captures exceptions (`task.map(_ => throw new RuntimeException("fail")) will be captured`. I just think it's not a good idea to rely on that as it is made for accidental exceptions that may leak into pure functions.

File: core/src/main/scala/org/http4s/MessageOps.scala [hunk 1]
```diff
@@ -87,7 +87,7 @@ trait MessageOps extends Any {
     * @return the `Task` which will generate the T
     */
   final def as[T](implicit decoder: EntityDecoder[T]): Task[T] =
-    attemptAs(decoder).fold(throw _, identity)
+    attemptAs(decoder).fold(Task.fail(_), _.pure[Task]).flatten
```
Comment (ChristopherDavenport) [inline core/src/main/scala/org/http4s/MessageOps.scala]: Is there a reason you favored this syntax over that used in [fs2 for fromAttempt](https://github.com/functional-streams-for-scala/fs2/blob/series/1.0/core/shared/src/main/scala/fs2/Task.scala#L369).

```scala
attemptAs(decoder).fold(Task.fail, Task.pure).flatten
```

Comment (wedens) [inline core/src/main/scala/org/http4s/MessageOps.scala]: No particular reason in this case. It's just more general syntax that works for any `Applicative`.

Comment (ChristopherDavenport) [inline core/src/main/scala/org/http4s/MessageOps.scala]: That seems like fine logic, and this code is completely understandable.

**Pull request merged**